CMakeLists.txt

```diff
@@ -0,0 +1,178 @@
+else ()
+    add_library(mem SHARED memutils/memory.h memutils/memory/malloc.c)
+endif ()
+
+# based on DEBUG and QUERY_MODE, transform CUDA_NVCC_FLAGS
+if (DEBUG STREQUAL "y")
+    set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -g -G --compiler-options='-g -ggdb'")
+else ()
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3")
+endif ()
+
+string(REPLACE " " ";" CUDA_XCOMPILER_FLAGS ${CMAKE_C_FLAGS})
+list(TRANSFORM CUDA_XCOMPILER_FLAGS PREPEND " -Xcompiler ")
+string(REPLACE ";" " " CUDA_XCOMPILER_FLAGS ${CUDA_XCOMPILER_FLAGS})
+# transform macro to prepend -D prefix
+list(TRANSFORM MARCROS PREPEND "-D ")
+set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} ${CUDA_XCOMPILER_FLAGS} ${GENCODE_FLAGS} ${MARCROS}")
+file(MAKE_DIRECTORY ${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
+file(MAKE_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
+
+cuda_add_library(algorithm SHARED
+        query/algorithm.hpp
+        query/algorithm.cu
+        query/binder.hpp
+        query/dimension_transform.cu
+        query/filter.cu
+        query/functor.cu
+        query/geo_intersects.cu
+        query/hash_lookup.cu
+        query/hll.cu
+        query/iterator.hpp
+        query/measure_transform.cu
+        query/scratch_space_transform.cu
+        query/sort_reduce.cu
+        query/transform.cu
+        query/transform.hpp
+        query/utils.cu
+        query/utils.hpp)
+
+
+################################
+# GTest
+################################
+set(GTEST_ROOT $ENV{GTEST_ROOT})
+link_directories(${GTEST_ROOT})
+enable_testing()
+################################
+# Unit Tests
+################################
+file(GLOB QUERY_UNITTEST_FILES query/*_unittest.cu)
+cuda_add_executable(all_unittest ${QUERY_UNITTEST_FILES})
+target_include_directories(all_unittest PRIVATE ${GTEST_ROOT}/include ${gtest_SOURCE_DIR})
+#add_dependencies(all_unittest algorithm)
+# Link test executable against gtest & gtest_main
+target_link_libraries(all_unittest gtest gtest_main algorithm pthread)
+add_test(all_unittest ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/all_unittest)
+
+###############################
+# golang
+###############################
+
+execute_process(COMMAND bash -c "find . -name '*.go' | grep -v \
+  -e vendor -e go-build \
+  -e build \
+  -e '.*/\\..*' \
+  -e '.*/_.*' \
+  -e '.*/mocks.*'"
+        OUTPUT_VARIABLE ALL_GO_SRC)
+string(REPLACE "\n" " " ALL_GO_SRC ${ALL_GO_SRC})
+
+execute_process(
+        COMMAND bash -c "git diff origin/master HEAD --name-only --diff-filter=AMR | grep -e \".*\\.go\""
+        OUTPUT_VARIABLE CHANGED_GO_SRC
+        )
+if (CHANGED_GO_SRC)
+    string(REPLACE "\n" " " CHANGED_GO_SRC ${CHANGED_GO_SRC})
+endif()
+
+
+execute_process(COMMAND bash -c "find .  -type f \
+    \\( -iname \\*.cu -o -iname \\*.h -o -iname \\*.hpp -o -iname \\*.c \\) \
+     | grep -v \
+  -e vendor -e go-build \
+  -e build \
+  -e '.*/\\..*' \
+  -e '.*/_.*' \
+  -e '.*/mocks.*'"
+        OUTPUT_VARIABLE ALL_C_SRC)
+string(REPLACE "\n" " " ALL_C_SRC ${ALL_C_SRC})
```

Comment:
optionally, pipe `find` output to  `| tr '\n' ' '`

Reply:
or, change the *lang-lint.sh scripts to handle both space and new line separated input

Reply:
we need those variable.e.g ALL_GO_SRC as dep files